tests/unit/utils/test_utils.py

```diff
@@ -993,3 +993,13 @@ def test_directory(self, tmp_path, caplog):
                 pass
         assert len(caplog.messages) == 1
         assert caplog.messages[0].startswith("Failed to delete tempfile")
+
+
+def test_parse_int_set_ranges():
+    lhs = sorted(utils.parse_int_set('2-5,7,15-17,12', []))
+    assert lhs == [2, 3, 4, 5, 7, 12, 15, 16, 17]
+
+
+def test_parse_int_set_wildcard():
+    lhs = sorted(utils.parse_int_set('*', [1, 2, 3]))
+    assert lhs == [1, 2, 3]
```

Comment:
A couple of suggestions:

- Please use [parametrization](https://docs.pytest.org/en/stable/parametrize.html#pytest-mark-parametrize) to turn those into one test function.
- Also see https://github.com/qutebrowser/qutebrowser/pull/2493/files#diff-c4e10bf2cd3f4c09bd62da3a402b29fd7ec2914cabc01e9de6cd2b13bec81780R928 for some more inspiration for possible text cases.
- Please also write a test using [hypothesis](https://hypothesis.readthedocs.io/en/latest/) to ensure the parsing doesn't raise any unexpected exceptions. For starters, a test just throwing `hypothesis.strategies.text()` at it will suffice (see e.g. `test_parse_duration_hypothesis` for an example) - but if you want to take this further (and learn a quite awesome tool), a more complex strategy actually generating those ranges would be nice (see `test_urlpattern_hypothesis` in `tests/unit/utils/test_urlmatch.py` for some inspiration).

Reply:
I think it's pretty solid now